thoth/package_extract/image.py

```diff
@@ -546,6 +546,47 @@ def get_image_size(path: str) -> int:
     return total_size
 
 
+def _get_cuda_version(path: str) -> dict:
+    """Get the cuda version."""
+    res = dict()
+    # gathering version from version.txt file
+    version_path = path + "/usr/local/cuda/version.txt"
+    if os.path.isfile(version_path):
+        with open(version_path, "r") as f:
+            output = f.read()
+            lines = output.split()
+            if len(lines) != 3:
+                _LOGGER.info(
+                    "Unable to detect CUDA version from usr/local/cuda/version.txt output: %r",
+                    output,
+                )
+            else:
+                res["/usr/local/cuda/version.txt"] = lines[-1]
```

Comment:
If we successfully passed the above condition, we are sure that the last string of the array is the version string, so we add it.